storage/disk/disk.go

```diff
@@ -235,6 +240,166 @@ func (db *Store) NewTransaction(ctx context.Context, params ...storage.Transacti
 	return newTransaction(xid, write, underlying, context, db.pm, db.partitions, db), nil
 }
 
+// Truncate implements the storage.Store interface.
```

Comment:
💭 Do we have some sort of mutex in place here to avoid this happening multiple times concurrently? Somewhere up the call stack, maybe? Perhaps this function would better have a comment re: only running once at a time, if we're assuming that the caller has that responsibility.

Reply:
BTW this could totally become the default procedure for backwards-incompatible partitioning changes... 🤔

Reply:
> Do we have some sort of mutex in place here to avoid this happening multiple times concurrently?

This function must be called with a transaction opened on the store. So when we call `NewTransaction` on the disk or in-memory store it takes the read/write lock based on the operation which is then released in the commit call.

So something like
1) Create new transaction on store
2) Called truncate
3) Call Commit/Abort

Reply:
I'll add a comment like `This method must be called within a transaction.`